tests/doc/test-levels.md

```diff
@@ -0,0 +1,150 @@
+# Test scopes
+
+## Overview
+
+This document is an analysis of testing scopes, current and proposed, in the SDK as they
+relate to [EPIC: Separate all SDK modules into standalone go modules](https://github.com/cosmos/cosmos-sdk/issues/11899).  Some tests have
+been written such that they introduce additional dependencies for the test phase only, and
+these must be addressed for the successful completion of Phases 3 and 4.
+
+## Unit tests
+
+Unit tests exercise the code contained in a single module (e.g. `/x/bank`) or package
```

Comment:
I personally think it's useful to describe two levels of unit tests:
* tests which exercise an atomic part of a module in isolation - in this case we might do fixture setup/mocking of other parts of the module
* tests which exercise a whole module's function with dependencies mocked, these are almost like integration tests in that they exercise many things together but still use mocks

I find this description of **illustrative** and **journey** scenarios from [The BDD Books – Formulation](https://leanpub.com/bddbooks-formulation) useful:
<img width="991" alt="image" src="https://user-images.githubusercontent.com/48918/182338724-509b480e-bde2-4035-8444-ec4d0b735f2b.png">

For better testing coverage, I think we would generally require more "illustrative scenarios" which cover _all_ cases and I have the sense we haven't really been disciplined enough there.